parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -519,7 +526,7 @@ public ParquetFileReader(
     this.conf = configuration;
     this.fileMetaData = fileMetaData;
     FileSystem fs = filePath.getFileSystem(configuration);
-    this.f = fs.open(filePath);
+    this.f = HadoopStreams.wrap(fs.open(filePath));
```

Comment:
eventually we could replace the "FileStatus file" parameter with something that provides a SeekableInputStream.
Possibly we just need length and Name to SeekableInputStream and call it ScannableFile ?
It is something that has a length (so that we can locate the footer) and where we and load predefined chunks from, in one go.